bootstrap/secret/insecure.go

```diff
@@ -146,3 +148,24 @@ func (p *InsecureProvider) ListSecretPaths() ([]string, error) {
 
 	return results, nil
 }
+
+// RegisteredSecretUpdatedCallback registers a callback for a secret.
+func (p *InsecureProvider) RegisteredSecretUpdatedCallback(path string, callback func(path string)) {
+	p.registeredSecretCallbacks[path] = callback
```

Comment:
This method as a subtle bug -- if path already has a callback, then this code will overwrite it.

Consequently, we either need to support multiple callbacks for a given path, or return an error if a callback is already registered.

As mentioned earlier, return a func() that if called will deregister the callback.